adapter/statsd/statsd_test.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsd
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/genproto/googleapis/rpc/status"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/duration"
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/cactus/go-statsd-client/statsd/statsdtest"
+	"istio.io/mixer/adapter/statsd/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/pkg/adapter/test"
+)
+
+func TestInvariants(t *testing.T) {
+	test.AdapterInvariants(Register, t)
+}
+
+func TestNewBuilder(t *testing.T) {
+	conf = &config.Params{
+		Address:                   "localhost:8125",
+		Prefix:                    "",
+		FlushDuration:             &duration.Duration{Seconds: 0, Nanos: int32(300 * time.Millisecond)},
+		FlushBytes:                512,
+		SamplingRate:              1.0,
+		MetricNameTemplateStrings: map[string]string{"a": `{{ .apiMethod "-" .responseCode }}`},
+	}
+	b := newBuilder()
```

Comment:
At present we have no way to pass parameters to newBuilder during registration. I don;t think these params belong here.